src/dm-screen/components/InitiativeOrderComponent/InitiativeOrderComponent.tsx

```diff
@@ -0,0 +1,302 @@
+      initiativeOrder.setSortValue(item.id, value);
+      handleUpdateInitiativeOrderInternal();
+      setInitiativeOrderState(initiativeOrder.getState());
+    }
+  };
+
+  const next = () => {
+    const initiativeOrder = getInitiativeOrder();
+    if (initiativeOrder) {
+      initiativeOrder.next();
+      handleUpdateInitiativeOrderInternal();
+      setInitiativeOrderState(initiativeOrder.getState());
+    }
+  };
+
+  const prev = () => {
+    const initiativeOrder = getInitiativeOrder();
+    if (initiativeOrder) {
+      initiativeOrder.prev();
+      handleUpdateInitiativeOrderInternal();
+      setInitiativeOrderState(initiativeOrder.getState());
+    }
+  };
+
+  const remove = () => {
+    const initiativeOrder = getInitiativeOrder();
+    if (initiativeOrder) {
+      initiativeOrder.remove(openId ?? '');
+      handleUpdateInitiativeOrderInternal();
+      setInitiativeOrderState(initiativeOrder.getState());
+    }
+    handleModalClose();
+  };
+
+  const reveal = () => {
+    const initiativeOrder = getInitiativeOrder();
+    if (initiativeOrder) {
+      initiativeOrder.reveal(openId ?? '');
+      handleUpdateInitiativeOrderInternal();
+      setInitiativeOrderState(initiativeOrder.getState());
+    }
+  };
+
+  const sort = () => {
+    const initiativeOrder = getInitiativeOrder();
+    if (initiativeOrder) {
+      initiativeOrder.sort();
+      handleUpdateInitiativeOrderInternal();
+      setInitiativeOrderState(initiativeOrder.getState());
+    }
+  };
+
+  const currentItem = initiativeOrderState?.items.find((item) => item.id === openId) ?? null;
+
+  return (
+    <>
+      <div className="initiative-order">
+        <div className="initiative-order-card-list">
+          {
+            initiativeOrderState?.items.map((item) => {
+              return (
+                <InitiativeCard
+                  active={item.id === initiativeOrderState?.currentId}
+                  imageSrc={item.imageSrc}
+                  key={item.id}
+                  name={item.name}
+                  onDoubleClick={() => {
+                    setIsOpen(true);
+                    setOpenId(item.id);
+                  }}
+                  onResourceAChange={(value) => {
+                    onResourceAChange(item, value);
+                  }}
+                  onResourceBChange={(value) => {
+                    onResourceBChange(item, value);
+                  }}
+                  onSortValueChange={(value) => {
+                    onSortValueChange(item, value);
+                  }}
+                  resourceA={item.resourceA}
+                  resourceB={item.resourceB}
+                  sortValue={item.sortValue}
+                  visibilityState={item.visibilityState}
+                />
+              );
+            })
+          }
+        </div>
+        <div className="initiative-order-card-button-container">
+          {
+            !playerView && (
+              <div>
+                <Button
```

Comment:
nit or future work: Can make a little reusable component for the buttons. Might not be worth it? 🤔